### PR TITLE
Export type definitions

### PR DESCRIPTION
**Pull request by samouri**
**summary**
Exports type definitions for usage in other TS based projects.

1. I didn't want to export type definitions for the `test` directories, but I still wanted to export them! Therefore I made a separate `tsconfig`
2. `outFile` will export all of the types in a single bundle, but _not_ as a module (only supports ambient)

 I named it `tsconfig-types`, although very happy to rename to something like. Possible idea: `tsconfig-gentypes`?

### CODE REVIEW

**Review by rcebulko — COMMENTED** [review]
What is this doing? Exporting only types but no code?

**Comment by samouri** [comment]
> What is this doing? Exporting only types but no code?

yes

File: package.json [hunk 1]
```diff
@@ -5,8 +5,10 @@
   "author": "The AMP HTML Authors",
   "license": "Apache-2.0",
   "main": "dist/index.js",
+  "types": "dist/types.d.ts",
   "files": [
-    "./dist/index.js"
+    "./dist/index.js",
```
Comment (jridgewell) [inline package.json:10]: Just export `"./dist/"`?

**Review by jridgewell — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (samouri) [reply package.json]: done

**Pull request merged**